crates/context/interface/src/transaction.rs

```diff
@@ -137,11 +137,15 @@ pub trait Transaction {
     ///
     /// While for transactions after Eip1559 it is minimum of max_fee and `base + max_priority_fee`.
     fn effective_gas_price(&self, base_fee: u128) -> u128 {
-        let max_fee = self.gas_price();
-        let Some(max_priority_fee) = self.max_priority_fee_per_gas() else {
-            return max_fee;
-        };
-        min(max_fee, base_fee.saturating_add(max_priority_fee))
+        if self.tx_type() == 0 || self.tx_type() == TransactionType::Eip2930 as u8 {
+            self.gas_price()
+        } else {
+            let max_fee = self.gas_price();
+            let Some(max_priority_fee) = self.max_priority_fee_per_gas() else {
+                return max_fee;
+            };
+            min(max_fee, base_fee.saturating_add(max_priority_fee))
+        }
```

Comment:
More readable code would be: 
```suggestion
        if self.tx_type() == 0 || self.tx_type() == TransactionType::Eip2930 as u8 {
            return self.gas_price();
        }
        
        // for EIP-1559 tx and onwards gas_price represents maximum price.
        let max_price = self.gas_price();
        let Some(max_priority_fee) = self.max_priority_fee_per_gas() else {
            return max_price;
        };
        min(max_fee, base_fee.saturating_add(max_priority_fee))
```

Reply:
Done